Cleanup and other non-fixes for two (and three)

albertvaka opened the pull request:
Best reviewed commit by commit, since one of the commits moves code around.

albertvaka committed:
Use !empty() instead of size() != 0

albertvaka committed:
Comment not true anymore?

free_integration_list doesn't exist

albertvaka committed:
Reorder code to keep same order in two and three

No code changes here!

albertvaka committed:
More explanatory message

albertvaka committed:
Check return value, like we do in three

albertvaka committed:
Remove unused var

albertvaka committed:
Explain why these AsString are okay

albertvaka committed:
Simplify code by decrefing at cleanup

Like we do everywhere else

KSerrania commented on six/two/two.cpp:
```diff
@@ -830,7 +838,6 @@ void Two::setIsExcludedCb(cb_is_excluded_t cb)
 // Python Helpers
 
 // get_integration_list return a list of every datadog's wheels installed. The
```
You forgot to remove the "The" at the end of the line which belonged to the sentence you removed.

albertvaka committed:
Typo

albertvaka committed:
Pass a non-null err string into PySequence_Fast

It should be the string used as error message if it fails. It is not an
out parameter and doesn't need to be freed.

albertvaka committed:
Use iterator's -> operator

truthbk commented on six/three/three.cpp:
```diff
@@ -89,7 +89,7 @@ bool Three::init()
 
     // Set PYTHONPATH
-    if (_pythonPaths.size()) {
+    if (!_pythonPaths.empty()) {
```
🍰

truthbk commented on six/common/builtins/aggregator.c:
```diff
@@ -95,7 +94,7 @@ static char **py_tag_to_c(PyObject *py_tags)
     }
 
-    py_tags_list = PySequence_Fast(py_tags, err); // new reference
+    py_tags_list = PySequence_Fast(py_tags, "py_tags is not a sequence"); // new reference
```
We should maybe start adding/consolidating error message strings, into header(s) for consistency (as a best practice). It's a bit of a chore and maybe beyond the scope of this PR, we could tackle that separately.

truthbk reviewed (dismissed):
This looks good to me, thank you for all this cleanup.

albertvaka committed:
Merge branch 'master' into albertvaka/two-minor-stuff
